br/pkg/lightning/backend/local/checksum.go

```diff
@@ -338,33 +339,21 @@ func (e *TiKVChecksumManager) checksumDB(ctx context.Context, tableInfo *checkpo
 	return nil, err
 }
 
-var retryGetTSInterval = time.Second
+var (
+	retryGetTSInterval    = time.Second
+	retryGetTSMaxDuration = time.Minute
+)
 
 // Checksum implements the ChecksumManager interface.
 func (e *TiKVChecksumManager) Checksum(ctx context.Context, tableInfo *checkpoints.TidbTableInfo) (*RemoteChecksum, error) {
 	tbl := common.UniqueTable(tableInfo.DB, tableInfo.Name)
 	var (
 		physicalTS, logicalTS int64
 		err                   error
-		retryTime             int
 	)
 	physicalTS, logicalTS, err = e.manager.pdClient.GetTS(ctx)
-	for err != nil {
-		if !pd.IsLeaderChange(errors.Cause(err)) {
```

Comment:
When meets unexpected error, we don't want to backoff, should fail-fast

And for tolerable error, we don't want a retry limit, 1 minute maybe not enough.

Reply:
I think the correct solution is not add another retry layers. We have too many retry layers now. API should know what specific error types are retryable and do not let caller bother the retry logic.

Reply:
Yes, I will check the error in backoffer

Reply:
IMO, for pd clients as a whole, caller should be able to give options as to what errors should not be retried and how many times.
Of course, the pd client should also provide a common default backoffer to deal with most cases, such as lightning. And this is exactly the starting point for maintaining a layer of backoff mechanism in pd client. PD's developer should maintain what kind of errors should be retried.